### PR TITLE
Ref #21 Extended project create form to provide id.

### PR DESCRIPTION
**Pull request by mariuszniemiec**
I did prety much very basic coverage, to not overthink myself.
However I really count on your feedback on this.
Only one intentional think I skipped is backend-side validation, still left out "for later" ;)

### CODE REVIEW

File: assets/js/pages/project/create.vue [hunk 1]
```diff
@@ -58,12 +72,27 @@
...
                 this.$v.$touch()
                 if (!this.$v.$invalid) {
-                    await api.create(this.project.name, this.project.description)
+                    await api.create(this.project.id, this.project.name, this.project.description)
                     this.$router.push('/')
                 }
             }
         },
+        watch: {
+            'project.id': function (newValue, oldValue) {
```
Comment (gbalcewicz) [inline assets/js/pages/project/create.vue]: If we are not using oldValue, it's not necessary to put it on argument list, it's javascript overall ;)

File: assets/js/pages/project/create.vue [hunk 2]
```diff
@@ -58,12 +72,27 @@
...
+                this.project.id = newValue.toUpperCase();
+            }
+        },
         computed: {
+            idErrors() {
+                const errors = []
+                if (!this.$v.project.id.$dirty) return errors
+                !this.$v.project.id.maxLength && errors.push('Id must be at most 80 characters long')
```
Comment (gbalcewicz) [inline assets/js/pages/project/create.vue]: Message mismatch the actual length constraints.

File: src/Controller/ProjectController.php [hunk 3]
```diff
@@ -32,7 +32,8 @@ public function create(Request $request, ProjectRepositoryInterface $repository)
     {
         //TODO: validation
         $data = json_decode($request->getContent(), true);
-        $project = new Project(ProjectId::new(), $data['name'], $data['description']);
+        $id = ProjectId::fromString($data['id']);
```
Comment (gbalcewicz) [inline src/Controller/ProjectController.php]: Beside the user input validation what's done in the TODO part :), in the ValueObject ProjectId from string we should add check for the pattern and throw an InvalidIdException (what is not implemented yet). ValueObject must be always valid.

File: assets/js/pages/project/create.vue (hunk 1 above)
Comment (mariuszniemiec) [reply assets/js/pages/project/create.vue]: Indeed I was too haste, I added both when I was checking how function works and forgot to cleanup.

**Pull request merged**